deltachat-ios/AppDelegate.swift

```diff
@@ -445,24 +450,54 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
 
     // MARK: - misc.
 
-    func openDatabase() {
-        guard let databaseLocation = DatabaseHelper(dcContext: dcContext).updateDatabaseLocation() else {
+    func migrateToDcAccounts() {
+        let dcContext = DcContext()
+
+        // first database migration to shared container, can be removed at some point
+        // implemented in April 2020 (https://github.com/deltachat/deltachat-ios/pull/612)
+        let databaseHelper = DatabaseHelper(dcLogger: dcContext.logger)
+        guard let databaseLocation = databaseHelper.updateDatabaseLocation() else {
             fatalError("Database could not be opened")
         }
-        logger.info("open: \(databaseLocation)")
         dcContext.openDatabase(dbFile: databaseLocation)
+        if dcContext.isConfigured() {
+            dcContext.closeDatabase()
+            if dcAccounts.migrateAccount(dbLocation: databaseLocation) == 0 {
```

Comment:
i assume that will result in accounts-manager having two accounts. as that is a bit unexpected at that point, maybe we should change core to not create a default account. in this case, if there is no account to migrate, we would just say add-account.

cc @link2xt

Reply:
from my point of view, we don't need the default account context. We can add one manually (programmatically), when we need it.

Reply:
This is fixed in https://github.com/deltachat/deltachat-core-rust/pull/2500

Reply:
i pulled current core and rebased this pr.